harness/determined/core/_context.py

```diff
@@ -39,20 +39,24 @@ def __init__(
         preempt: Optional[core.PreemptContext] = None,
         train: Optional[core.TrainContext] = None,
         searcher: Optional[core.SearcherContext] = None,
+        profiler: Optional[core.ProfilerContext] = None,
     ) -> None:
         self.checkpoint = checkpoint
         self.distributed = distributed or core.DummyDistributedContext()
         self.preempt = preempt or core.DummyPreemptContext(self.distributed)
         self.train = train or core.DummyTrainContext()
         self.searcher = searcher or core.DummySearcherContext(self.distributed)
+        self.profiler = profiler
```

Comment:
prolly you'll want a DummyProfilerContext somewhere, so `core_context.profiler.*` isn't broken in these cases.

Reply:
done